### ISSUE TITLE
Exporting zod types from zValidator

### ISSUE THREAD

**Issue by thijserven** [honojs/middleware#154]
So I'm trying to implement @hono/zod-validator in my project. After following the README.md instruction everything works well and `req.valid('json');` is typed with using the provided zod schema.

But I like to have my handler functions is a separate typescript file. So how can I make sure that my zod schema types are available in my external handler function?

This is how I have it set up now:

```
// route.ts
app.post('/', zValidator('json', ...zodSchema), myHandler); // Typescript not complaining
```

```
// handler.ts
export function myHandler(c: Context) {
  const {someInput} = c.req.valid('json'); // [TS error]: Argument of type 'string' is not assignable to parameter of type 'never'.  
  return c.text('Test');
}
```

Is there any way I can import the zod schema types from the zValidator function in the Context type of myHandler?

**Comment by yusukebe** [comment]
Hi @thijserven!

Thank you for bringing up the issue.

For accurate type inference, I recommend defining the path, validator, and handler all at once:

```ts
app.post('/', zValidator('json', schema), (c) => {})
```

But, I know that some users prefer the separation, as you've mentioned. So, I'm struggling to resolve this "types" matter.

Using a custom context type seems like a good approach. I think this wouldn't just benefit the Zod Validator but could be a more general solution.

I'll consider it. For now, let's keep this issue open.

**Comment by coffeechain23** [comment]
@yusukebe I have upgraded the package and I see the same issue. I will have to write custom validator everywhere? 🐙

**Comment by yusukebe** [comment]
@coffeechain23 

No. As noted above, you can use the zValidator with the following syntax. Is this not enough?

```ts
app.post('/', zValidator('json', schema), (c) => {})
```

**Comment by mhio** [comment]
I've run into a similar issue extracting the zValidator `hook` handler function out to a generic function to build specific validation errors across an app.  
```
function convertZodError(result) {
  if (result.success) return
  // build custom response based on result.error
}
```
I need to give the hook the `Hook<z.infer<T>, E, P>` signature that the validator middleware creates to infer the `result` types (and the context but I'm not using that at the moment).

**Comment by KaelWD** [comment]
https://hono.dev/guides/best-practices#don-t-make-controllers-when-possible

**Comment by eprig** [comment]
Is this still being considered? I just came across the same issue
